Source/Turbo Navigator/TurboNavigator.swift

```diff
@@ -205,3 +216,89 @@ extension TurboNavigator: TurboWKUIDelegate {
+        /// it will trigger various bridge messages since the web view will be added to the window and call all the connect() methods.
+        ///
+        /// We don't want to reload a view controller not on screen, since that can have unwanted
+        /// side-effects for the next visit (like showing the wrong bridge components). We can't just
+        /// check if the view controller is visible, since it may be further back in the stack of a navigation controller.
+        /// Seeing if there is a parent was the best solution I could find.
+        guard let viewController = session.activeVisitable?.visitableViewController,
+              viewController.parent != nil else {
+            return
+        }
+        
+        if appInBackground {
+            /// Don't reload the web view if the app is in the background.
+            /// Instead, save the session in `backgroundTerminatedWebViewSessions`
+            /// and reload it when the app is back in foreground.
+            backgroundTerminatedWebViewSessions.append(session)
+            return
+        }
+        
+        reload(session)
+    }
+    
+    private func reload(_ session: Session) {
+        session.reload()
+    }
+    
+    /// Inspects the provided session to handle terminated web view process and reloads or recreates the web view accordingly.
+    ///
+    /// - Parameter session: The session to inspect.
+    ///
+    /// This method checks if the web view associated with the session has terminated in the background. 
+    /// If so, it removes the session from the list of background terminated web view processes, reloads the session, and returns.
+    /// If the session's topmost visitable URL is not available, the method returns without further action.
+    /// If the web view's content process state is non-recoverable/terminated, it recreates the web view for the session.
+    private func inspect(_ session: Session) {
+        if let index = backgroundTerminatedWebViewSessions.firstIndex(where: { $0 === session }) {
+            backgroundTerminatedWebViewSessions.remove(at: index)
+            reload(session)
+            return
+        }
+        
+        guard let _ = session.topmostVisitable?.visitableURL else {
+            return
+        }
+        
+        session.webView.queryWebContentProcessState { [weak self] state in
+            guard case .terminated = state else { return }
+            self?.recreateWebView(for: session)
+        }
+    }
+    
+    /// Recreates the web view and session for the given session and performs a `replace` visit.
+    ///
+    /// - Parameter session: The session to recreate.
+    private func recreateWebView(for session: Session) {
+        guard let _ = session.activeVisitable?.visitableViewController,
+              let url = session.activeVisitable?.visitableURL else { return }
+        
+        let newSession = Session(webView: Turbo.config.makeWebView())
```

Comment:
What about when someone uses their own session, via `TurboNavigator.init(session:modalSession:delegate:)`?

Reply:
I forgot to mention this. Thanks for bringing this up! 
Since we already provide the option to configure a custom web view via `Turbo.config.makeCustomWebView`, `init(session:modalSession:delegate:)` seems redundant.
I suggest we make it internal to the library.

Reply:
I'm all for that change! I've never used that and always let Turbo Navigator configure the web view for me. 👍

Reply:
Agree. Moved 👍🏻